viz_scripts/generic_metrics.ipynb

```diff
@@ -119,193 +119,366 @@
   },
   {
    "cell_type": "markdown",
-   "id": "modified-skiing",
+   "id": "6d44c87e",
    "metadata": {},
    "source": [
-    "## Generic Metrics"
+    "## Collect Data from Database for Sensed Metrics"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "55ec383b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "expanded_ct_sensed, file_suffix_sensed, quality_text_sensed, debug_df_sensed = scaffolding.load_viz_notebook_sensor_inference_data(year,\n",
+    "                                                                            month,\n",
+    "                                                                            program,\n",
+    "                                                                            include_test_users,\n",
+    "                                                                            sensed_algo_prefix)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8ecfd343",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a combined debug df from Generic and Sensed Metrics\n",
+    "merged_debug_df = debug_df.combine_first(debug_df_sensed)"
```

Comment:
We are using debug_df to create a generate_missing_plot() to show a table explaining why the chart(s) could not be generated. 
Earlier we had `debug_df` and `debug_df_sensed` separately in different charts. Now, we have a single consolidated chart to represent both of these. 
Let us understand what does `debug_df`, `debug_df_sensed` and `merged_debug_df` look now:

```
                                             value
year                                           NaN
month                                          NaN
Registered_participants                       20.0
Participants_with_at_least_one_trip           19.0
Participant_with_at_least_one_labeled_trip    17.0
Trips_with_at_least_one_label               3441.0
Trips_with_mode_confirm_label               3441.0
Trips_with_trip_purpose_label               3441.0


sensed 

                                      value
year                                    NaN
month                                   NaN
Registered_participants                20.0
Participants_with_at_least_one_trip    19.0
Number of trips                      7013.0


Combined df
                                             value
Number of trips                             7013.0
Participant_with_at_least_one_labeled_trip    17.0
Participants_with_at_least_one_trip           19.0
Registered_participants                       20.0
Trips_with_at_least_one_label               3441.0
Trips_with_mode_confirm_label               3441.0
Trips_with_trip_purpose_label               3441.0
month                                          NaN
year                                           NaN
```

- `debug_df_sensed` is a subset of the `debug_df` except `debug_df_sensed` has `Number of trips`.
- If there would be `null` value in `debug_df` while `debug_df_sensed`, it would likely replace the null values with values from the second df. 
But with this approach I am trying to leverage the aspect of `combine_first` such that 
```
The row and column indexes of the resulting DataFrame will be the union of the two.
The resulting dataframe contains the ‘first’ dataframe values and overrides the second one values where both
first.loc[index, col] and second.loc[index, col] are not missing values, upon calling first.combine_first(second).
```
Thereby, we will have a single consolidated `debug df` which will represent the values in the tabular format whenever we encounter any exception(s).

Reply:
That makes sense now, thank you for explaining!